### PR TITLE
feat: sdk 1333 migration of context user id session id opt out status from

### PR DESCRIPTION
**Pull request by itsdebs**
feat: migrate ids from v1 to v2 if source ids match

### CODE REVIEW

File: android/src/main/java/com/rudderstack/android/RudderAnalytics.kt [hunk 1]
```diff
@@ -147,21 +145,44 @@ private val infrastructurePlugins
...
 private fun Analytics.startup() {
     addPlugins()
     associateStates()
-    androidStorage.let {
-        contextState?.update(it.context)
-    }
+//    initializeSessionManagement()
+}
+
+/*private fun Analytics.reinstate() {
```
Comment (ChryssaAliferi) [inline android/src/main/java/com/rudderstack/android/RudderAnalytics.kt:158]: Let's remove this code if not used anymore. I can see that Reinstate() is now a plugin 😄

File: android/src/main/java/com/rudderstack/android/RudderAnalytics.kt [hunk 2]
```diff
@@ -147,21 +145,44 @@ private val infrastructurePlugins
...
+        }
+        val newUserId =
+            currentConfigurationAndroid?.userId ?: androidStorage.userId ?: androidStorage.v1UserId
+        applyConfiguration {
+            if (this is ConfigurationAndroid) copy(
+                anonymousId = newAnonId, userId = newUserId
+            )
+            else this
+        }
+    }
```
Comment (ChryssaAliferi) [inline android/src/main/java/com/rudderstack/android/RudderAnalytics.kt:183]: 🧹

File: android/src/main/java/com/rudderstack/android/internal/RudderPreferenceManager.kt [hunk 3]
```diff
@@ -109,12 +110,12 @@ internal class RudderPreferenceManager(application: Application,
...
-    }
-
-    val optStatus: Boolean
-        get() = preferences.getBoolean(RUDDER_OPT_STATUS_KEY.key, false)
+//    fun saveOptStatus(optStatus: Boolean) {
+//        preferences.edit().putBoolean(RUDDER_OPT_STATUS_KEY.key, optStatus).apply()
+//    }
+//
+//    val optStatus: Boolean
+//        get() = preferences.getBoolean(RUDDER_OPT_STATUS_KEY.key, false)
```
Comment (ChryssaAliferi) [inline android/src/main/java/com/rudderstack/android/internal/RudderPreferenceManager.kt:118]: I guess this is going to be used in the next small feature. 👀 
If so let's keep it!

File: android/src/test/java/com/rudderstack/android/internal/plugins/ReinstatePluginTest.kt [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+    }
+
+    @Test
+    fun `intercept should return original message when not reinstated`() {
+        val interceptedMessage = plugin.intercept(chain)
+        assertEquals(chain.originalMessage, interceptedMessage)
+    }
+
+    //    @Test
+//    fun `post reinstate messages should not change when `
```
Comment (ChryssaAliferi) [inline android/src/test/java/com/rudderstack/android/internal/plugins/ReinstatePluginTest.kt:104]: 🧹

File: android/src/test/java/com/rudderstack/android/internal/plugins/ReinstatePluginTest.kt [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+            )
+        }
+        dummyMessage = TrackMessage.create("dummyEvent", RudderUtils.timeStamp)
+        chain = mock<Plugin.Chain>()
+        config = RudderServerConfig(
+            source = RudderServerConfig.RudderServerConfigSource(
+                "testSourceId", isSourceEnabled = true
+            )
+        )
+//        context = mock(Context::class.java)
```
Comment (ChryssaAliferi) [inline android/src/test/java/com/rudderstack/android/internal/plugins/ReinstatePluginTest.kt:75]: 🧹

File: android/src/test/java/com/rudderstack/android/internal/plugins/ReinstatePluginTest.kt [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+@RunWith(AndroidJUnit4::class)
+@Config(sdk = [29])
+class ReinstatePluginTest {
+    private lateinit var analytics: Analytics
+    private lateinit var chain: Plugin.Chain
+    private lateinit var config: RudderServerConfig
+    private lateinit var androidStorage: AndroidStorage
+    private lateinit var mockControlPlane: ConfigDownloadService
+
+    //    private lateinit var context: Context
```
Comment (ChryssaAliferi) [inline android/src/test/java/com/rudderstack/android/internal/plugins/ReinstatePluginTest.kt:47]: 🧹

File: android/src/main/java/com/rudderstack/android/utilities/SessionUtils.kt [hunk 7]
```diff
@@ -93,9 +93,10 @@ internal fun Analytics.startSessionIfNeeded() {
     }
 }
 
-internal fun Analytics.initializeSessionManagement() {
-    val savedSessionId = androidStorage.sessionId
-    val lastActiveTimestamp = androidStorage.lastActiveTimestamp
+internal fun Analytics.initializeSessionManagement(savedSessionId: Long? = null,
+        lastActiveTimestamp: Long? = null) {
+//    val savedSessionId = androidStorage.sessionId
```
Comment (ChryssaAliferi) [inline android/src/main/java/com/rudderstack/android/utilities/SessionUtils.kt:98]: 🧹

**Review by ChryssaAliferi — APPROVED** [review]

File: android/src/main/java/com/rudderstack/android/internal/RudderPreferenceManager.kt (hunk 3 above)
Comment (itsdebs) [reply android/src/main/java/com/rudderstack/android/internal/RudderPreferenceManager.kt:118]: Removing this, will be added in different PR

**Pull request merged**